### PR TITLE
Implement most Rogue, Warlock, and neutral cards

### PR DESCRIPTION
**Pull request by NightKev**
Cards implemented:
* Rogue: Bladed Cultist, Southsea Squidface, Shadow Strike, Journey Below, Undercity Huckster, and "Xaril, Poisoned Mind" + toxins
* Warlock: Darkshire Librarian, Darkshire Councilman, Possessed Villager, Spreading Madness, and DOOM!
* Neutral commons: Aberrant Berserker, Infested Tauren, Spawn of N'Zoth, Polluted Hoarder, Cult Apothecary and Nerubian Prophet
* Neutral rares: Blackwater Pirate, Eater of Secrets, Corrupted Healbot, Corrupted Seer, and Midnight Drake
* Neutral epics: Twilight Summoner, Cyclopian Horror, Validated Doomsayer, and Ancient Harbinger
* Neutral legendaries: "Mukla, Tyrant of the Vale", "Hogger, Doom of Elwynn", "Nat, the Darkfisher", and "Deathwing, Dragonlord"

Tests implemented for:
* Rogue: Journey Below, Undercity Huckster, and "Xaril, Poisoned Mind"
* Warlock: DOOM!
* Neutral commons: Cult Apothecary and Nerubian Prophet
* Neutral rares: Blackwater Pirate and Eater of Secrets
* Neutral epics: Validated Doomsayer and Ancient Harbinger
* Neutral legendaries: "Deathwing, Dragonlord"

### CODE REVIEW

File: tests/test_wog.py [hunk 1]
```diff
@@ -180,6 +180,18 @@ def test_cult_apothecary():
 	assert game.player1.hero.health == 28
 
 
+def test_deathwing_dragonlord():
+	game = prepare_empty_game()
```
Comment (jleclanche) [inline tests/test_wog.py]: Use `prepare_game()` and discard player1's hand.

File: tests/test_wog.py [hunk 2]
```diff
@@ -180,6 +180,18 @@ def test_cult_apothecary():
...
+	deathwing = game.player1.give("OG_317")
+	game.player1.give("EX1_572")
+	game.player1.give("EX1_284")
+	game.player1.give(WISP)
+	deathwing.play()
+	deathwing.destroy()
+	assert len(game.player1.hand) == 1
+	assert len(game.player1.field) == 2
```
Comment (jleclanche) [inline tests/test_wog.py]: Stuff those cards in variables and compare the hand and field to lists, eg: `game.player1.hand == [one, two]`.

Incidentally this will check the ordering of the minions on the field. Cross your fingers we're getting that right :)

**Comment by NightKev** [comment]
Deathwing test updated.

File: tests/test_wog.py [hunk 3]
```diff
@@ -596,3 +596,13 @@ def test_xaril_toxins():
...
+
+
+def test_yshaarj_rage_unbound():
+	game = prepare_game()
+	yshaarj = game.player1.give("OG_042")
+	yshaarj.play()
+	assert len(game.player1.deck) == 26
+	game.end_turn()
+	assert len(game.player1.field) == 2
+	assert len(game.player1.deck) == 25
```
Comment (jleclanche) [inline tests/test_wog.py]: This is a flaky test which will fail if there's no minions in the deck. Prepare an empty game and shuffle a specific minion and spell in the deck.

Make sure to also test with an empty deck.

File: fireplace/cards/wog/rogue.py [hunk 4]
```diff
@@ -1,4 +1,5 @@
 from ..utils import *
+from .toxins import *
```
Comment (jleclanche) [inline fireplace/cards/wog/rogue.py]: No, this should be in __init__.

File: tests/test_wog.py [hunk 5]
```diff
@@ -456,3 +456,37 @@ def test_wisps_of_the_old_gods():
...
+def test_xaril_poisoned_mind():
+	game = prepare_empty_game()
+	xaril = game.player1.give("OG_080")
+	xaril.play()
+	assert game.player1.hand[0] in ("OG_080b", "OG_080c", "OG_080d", "OG_080e", "OG_080f")
+	game.player1.give("DS1_185").play(target=xaril)
+	assert game.player1.hand[1] in ("OG_080b", "OG_080c", "OG_080d", "OG_080e", "OG_080f")
+
+
+def test_xaril_toxins():
```
Comment (jleclanche) [inline tests/test_wog.py]: You're not actually testing any of the toxins though.

Also if you do, they should just have their own test. But it's notnecessary.

Comment (NightKev) [inline tests/test_wog.py]: Should I take the Xaril tests out (or combine them)?

File: tests/test_wog.py [hunk 6]
```diff
@@ -426,6 +426,16 @@ def test_thistle_tea():
...
 
 
+def test_undercity_huckster():
+	game = prepare_empty_game()
+	undercity_huckster = game.player1.give("OG_330")
+	undercity_huckster.play()
+	arcane_shot = game.player1.give("DS1_185")
+	arcane_shot.play(target=undercity_huckster)
+	assert game.player1.hand[0].card_class == game.player2.hero.card_class
+	assert game.player1.hand[0].type != CardType.HERO
```
Comment (jleclanche) [inline tests/test_wog.py]: Drop that assertiot

File: tests/test_wog.py [hunk 7]
```diff
@@ -140,6 +140,20 @@ def test_chogall_cannot_pay_health():
...
 
+def test_cult_apothecary():
+	game = prepare_game()
+	cult_apothecary = game.player1.give("OG_295")
+	game.end_turn()
+	game.player2.give(WISP).play()
+	game.player2.give(WISP).play()
+	game.player2.give(WISP).play()
+	game.player2.give(WISP).play()
+	game.player2.give("EX1_279").play(target=game.player1.hero)
```
Comment (jleclanche) [inline tests/test_wog.py]: Use the DAMAGE_5 token card

File: tests/test_wog.py [hunk 8]
```diff
@@ -140,6 +140,20 @@ def test_chogall_cannot_pay_health():
...
+	game.end_turn()
+	cult_apothecary.play()
+	assert game.player1.hero.health == 28
```
Comment (jleclanche) [inline tests/test_wog.py]: Add a lightwarden to this test so you can double check the heal only triggers once.

File: tests/test_wog.py [hunk 9]
```diff
@@ -26,6 +26,15 @@ def test_addled_grizzly():
 	assert wisp.atk == wisp.health == 3
 
 
+def test_blackwater_pirate():
+	game = prepare_game()
+	game.player1.give("OG_322").play()
+	reaper = game.player1.give("CS2_112")
```
Comment (jleclanche) [inline tests/test_wog.py]: Add a minion and a spell to the hand as well to check they don't get reduced.

File: tests/test_wog.py [hunk 10]
```diff
@@ -188,6 +188,20 @@ def test_doom():
...
+	eater_of_secrets = game.player1.give("OG_254")
+	game.end_turn()
+	game.player2.give("EX1_379").play()
+	game.player2.give("EX1_609").play()
+	game.player2.give("EX1_294").play()
+	game.end_turn()
+	eater_of_secrets.play()
+	assert eater_of_secrets.atk == 5
+	assert eater_of_secrets.health == 7
+	assert len(game.player2.secrets) == 0
```
Comment (jleclanche) [inline tests/test_wog.py]: `assert not game.player2.secrets`

File: tests/test_wog.py [hunk 11]
```diff
@@ -497,6 +497,21 @@ def test_undercity_huckster():
 	assert game.player1.hand[0].type != CardType.HERO
 
 
+def test_validated_doomsayer():
+	game = prepare_game()
+	validated_doomsayer = game.player1.give("OG_200")
+	validated_doomsayer.play()
+	assert validated_doomsayer.atk == 0
+	game.end_turn(); game.end_turn()
```
Comment (jleclanche) [inline tests/test_wog.py]: Add a check during the opponent's turn as well.

File: tests/test_wog.py [hunk 12]
```diff
@@ -26,6 +26,23 @@ def test_addled_grizzly():
 	assert wisp.atk == wisp.health == 3
 
 
+def test_ancient_harbinger():
+	game = prepare_empty_game()
+	game.player1.discard_hand()
```
Comment (jleclanche) [inline tests/test_wog.py]: You don't need this for empty games.

File: tests/test_wog.py [hunk 13]
```diff
@@ -12,7 +12,6 @@ def test_a_light_in_the_darkness():
 		card.clear_buffs()
 		basehp = card.health
 		assert buffhp == basehp + 1
-		# TODO: put a test for card class here, once it's implemented
```
Comment (jleclanche) [inline tests/test_wog.py]: You don't get to remove a todo if you didn't do it :)
